### ISSUE TITLE
[1.12.2-rc1] devicemapper: Error running deviceResume dm_task_run failed

### ISSUE THREAD

**Issue by JamesJJ** [moby/moby#27002]
**Description**

Starting containers with `docker-compose` sometimes shows `devicemapper: Error running deviceResume dm_task_run failed`, container fails to start, compose's docker API call times out.

**Steps to reproduce the issue:**

```
docker-compose pull
sleep 2
date -u
lsblk
do
cker-compose up --no-build --force-recreate --no-color -d
sleep 2
date -u
```

**Describe the results you received:**

```
...
Pulling logstack-client (aa-dev-registry-usw2.domain.com/ops/logstack_client:20160919a)...
20160919a: Pulling from ops/logstack_client
Digest: sha256:7f30b462c39b3bb13b46e323fc449d12fc1f1b36b1028a865a56b35287bcff98
Status: Image is up to date for aa-dev-registry-usw2.domain.com/ops/logstack_client:20160919a

Thu Sep 29 04:12:36 UTC 2016

NAME                             MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
xvda                             202:0    0    8G  0 disk 
└─xvda1                          202:1    0    8G  0 part /
xvdf                             202:80   0   50G  0 disk 
├─docker_lvm_vg-docker_lvm_tmeta 253:0    0  1.5G  0 lvm  
│ └─docker_lvm_vg-docker_lvm     253:2    0 46.5G  0 lvm  
└─docker_lvm_vg-docker_lvm_tdata 253:1    0 46.5G  0 lvm  
  └─docker_lvm_vg-docker_lvm     253:2    0 46.5G  0 lvm  

Creating aaaaaaaaaaa1_aa-app-aa_1
Creating aaaaaaaaaaa1_aa-aaaaaa-camp_1
Creating aaaaaaaaaaa1_aa-service_1
Creating aaaaaaaaaaa1_logstack-client_1
Creating aaaaaaaaaaa1_monitor-docker_1
Creating aaaaaaaaaaa1_sgngx_1

ERROR: for aa-service  UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)

ERROR: for aa-app-aa  UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)

ERROR: for logstack-client  Cannot create container for service logstack-client: devicemapper: Error running deviceResume dm_task_run failed

ERROR: for aa-aaaaaa-camp  UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)
An HTTP request took too long to complete. Retry with --verbose to obtain debug information.
If you encounter this issue regularly because of slow network conditions, consider setting COMPOSE_HTTP_TIMEOUT to a higher value (current value: 60).

Thu Sep 29 04:13:42 UTC 2016

```

**Describe the results you expected:**

No error, 6 containers running.

**Additional information you deem important (e.g. issue happens only occasionally):**
- Does not happen every time

```
journalctl -u docker.service | grep -A 20 -B 20 'deviceResume dm_task_run'
Sep 29 04:11:45 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:11:45.088319261Z" level=info msg="containerd: e4b235b41d7404637b6d0948935c717b9cd4ded7909c53223bd3f1ea3d984d54:32bea2a693957e628b9d7ab1cec5617cdc2d9bce949d776b0aab75515d79a544 (pid 14821) has become an orphan, killing it"
Sep 29 04:11:49 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:11:49.125122684Z" level=warning msg="containerd: unable to save e4b235b41d7404637b6d0948935c717b9cd4ded7909c53223bd3f1ea3d984d54:48db2fcd7119727c01a551d0e48e6d18d9ac49b14f0a9573ee588c28313d8c25 starttime: open /proc/14840/stat: no such file or directory"
Sep 29 04:11:49 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:11:49.12528892Z" level=info msg="containerd: e4b235b41d7404637b6d0948935c717b9cd4ded7909c53223bd3f1ea3d984d54:48db2fcd7119727c01a551d0e48e6d18d9ac49b14f0a9573ee588c28313d8c25 (pid 14840) has become an orphan, killing it"
Sep 29 04:11:53 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:11:53.22811764Z" level=warning msg="containerd: unable to save e4b235b41d7404637b6d0948935c717b9cd4ded7909c53223bd3f1ea3d984d54:4410d49ff2c88f5da4bc8078370e3a07f72d3c631b9ea641d286cff51ada2b66 starttime: open /proc/15005/stat: no such file or directory"
Sep 29 04:11:53 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:11:53.229395656Z" level=info msg="containerd: e4b235b41d7404637b6d0948935c717b9cd4ded7909c53223bd3f1ea3d984d54:4410d49ff2c88f5da4bc8078370e3a07f72d3c631b9ea641d286cff51ada2b66 (pid 15005) has become an orphan, killing it"
Sep 29 04:12:05 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:05.572177808Z" level=warning msg="containerd: unable to save e4b235b41d7404637b6d0948935c717b9cd4ded7909c53223bd3f1ea3d984d54:b512891fd13f4079c5fd1db7c9a01a1d269975c9e214b873c2a372b7749ac92c starttime: open /proc/15179/stat: no such file or directory"
Sep 29 04:12:05 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:05.572362513Z" level=info msg="containerd: e4b235b41d7404637b6d0948935c717b9cd4ded7909c53223bd3f1ea3d984d54:b512891fd13f4079c5fd1db7c9a01a1d269975c9e214b873c2a372b7749ac92c (pid 15179) has become an orphan, killing it"
Sep 29 04:12:09 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:09.625834018Z" level=warning msg="containerd: unable to save e4b235b41d7404637b6d0948935c717b9cd4ded7909c53223bd3f1ea3d984d54:b2d551214c81fc4453308e507a0576046f22872fa9ae755a3c451338dd486621 starttime: open /proc/15201/stat: no such file or directory"
Sep 29 04:12:09 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:09.626002949Z" level=info msg="containerd: e4b235b41d7404637b6d0948935c717b9cd4ded7909c53223bd3f1ea3d984d54:b2d551214c81fc4453308e507a0576046f22872fa9ae755a3c451338dd486621 (pid 15201) has become an orphan, killing it"
Sep 29 04:12:22 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:22.220135119Z" level=warning msg="containerd: unable to save e4b235b41d7404637b6d0948935c717b9cd4ded7909c53223bd3f1ea3d984d54:d62017caa5fc6324b58b0956128d0118775b286757e6b41df26518cdf690fb5d starttime: open /proc/15301/stat: no such file or directory"
Sep 29 04:12:22 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:22.220297378Z" level=info msg="containerd: e4b235b41d7404637b6d0948935c717b9cd4ded7909c53223bd3f1ea3d984d54:d62017caa5fc6324b58b0956128d0118775b286757e6b41df26518cdf690fb5d (pid 15301) has become an orphan, killing it"
Sep 29 04:12:25 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:25.787594929Z" level=warning msg="libcontainerd: container 3ec15979924afd103805f3aab2b2f66412d7f3b5c96dda6bf4e0d9c87e009a65 restart canceled"
Sep 29 04:12:26 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:26.046742652Z" level=warning msg="libcontainerd: container 41c535dc29f0c04acf751eee41a4e62c8803ff01640baa02b64f4da77636513d restart canceled"
Sep 29 04:12:26 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:26.166366547Z" level=warning msg="libcontainerd: container e4b235b41d7404637b6d0948935c717b9cd4ded7909c53223bd3f1ea3d984d54 restart canceled"
Sep 29 04:12:26 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:26.300891601Z" level=warning msg="libcontainerd: container 04fecbd2f8b85151f993c99d24f7bc2514e8871f1fbe2078f40e376f3638997b restart canceled"
Sep 29 04:12:26 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:26.430583296Z" level=warning msg="libcontainerd: container 4b69244ddf2b2b0dc43ad5c55e4ebe0e823072fd31dc3329bb54f02929fa3136 restart canceled"
Sep 29 04:12:26 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:26.552263926Z" level=warning msg="libcontainerd: container 33095cb862ab281cfa0d5b340b85c038eb79efbe48d8d478233b97ca13338e35 restart canceled"
Sep 29 04:12:38 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:38Z" level=info msg="Firewalld running: false"
Sep 29 04:12:38 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:38Z" level=info msg="Firewalld running: false"
Sep 29 04:12:39 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:39Z" level=info msg="Firewalld running: false"
Sep 29 04:12:39 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:39.292119674Z" level=error msg="Handler for POST /v1.22/containers/create returned error: devicemapper: Error running deviceResume dm_task_run failed"
Sep 29 04:12:40 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:40Z" level=info msg="Firewalld running: false"
Sep 29 04:12:43 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:43.363585317Z" level=warning msg="Health check error: rpc error: code = 2 desc = containerd: container not found"
Sep 29 04:12:43 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:43Z" level=info msg="Firewalld running: false"
Sep 29 04:12:44 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:44Z" level=info msg="Firewalld running: false"
Sep 29 04:12:47 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:47.377566596Z" level=warning msg="Health check error: rpc error: code = 2 desc = containerd: container not found"
Sep 29 04:12:51 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:51.387568101Z" level=warning msg="Health check error: rpc error: code = 2 desc = containerd: container not found"
Sep 29 04:12:55 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:55.402482555Z" level=warning msg="Health check error: rpc error: code = 2 desc = containerd: container not found"
Sep 29 04:12:59 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:12:59.411513338Z" level=warning msg="Health check error: rpc error: code = 2 desc = containerd: container not found"
Sep 29 04:13:03 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:13:03.413786450Z" level=warning msg="Health check error: rpc error: code = 2 desc = containerd: container not found"
Sep 29 04:13:07 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:13:07.416103784Z" level=warning msg="Health check error: rpc error: code = 2 desc = containerd: container not found"
Sep 29 04:13:11 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:13:11.418392658Z" level=warning msg="Health check error: rpc error: code = 2 desc = containerd: container not found"
Sep 29 04:13:15 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:13:15.420710167Z" level=warning msg="Health check error: rpc error: code = 2 desc = containerd: container not found"
Sep 29 04:13:19 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:13:19.422918149Z" level=warning msg="Health check error: rpc error: code = 2 desc = containerd: container not found"
Sep 29 04:13:23 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:13:23.425256705Z" level=warning msg="Health check error: rpc error: code = 2 desc = containerd: container not found"
Sep 29 04:13:27 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:13:27.427550728Z" level=warning msg="Health check error: rpc error: code = 2 desc = containerd: container not found"
Sep 29 04:13:31 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:13:31.429851058Z" level=warning msg="Health check error: rpc error: code = 2 desc = containerd: container not found"
Sep 29 04:13:35 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:13:35.432155541Z" level=warning msg="Health check error: rpc error: code = 2 desc = containerd: container not found"
Sep 29 04:13:39 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:13:39.434446952Z" level=warning msg="Health check error: rpc error: code = 2 desc = containerd: container not found"
Sep 29 04:13:43 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:13:43.436742568Z" level=warning msg="Health check error: rpc error: code = 2 desc = containerd: container not found"
Sep 29 04:13:47 ip-10-10-12-33 dockerd[5092]: time="2016-09-29T04:13:47.439053168Z" level=warning msg="Health check error: rpc error: code = 2 desc = containerd: container not found"
```

**Output of `docker version`:**

```
Client:
 Version:      1.12.2-rc1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   45bed2c
 Built:        
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.2-rc1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   45bed2c
 Built:        
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 5
 Running: 2
 Paused: 0
 Stopped: 3
Images: 8
Server Version: 1.12.2-rc1
Storage Driver: devicemapper
 Pool Name: docker_lvm_vg-docker_lvm
 Pool Blocksize: 524.3 kB
 Base Device Size: 8.59 GB
 Backing Filesystem: xfs
 Data file: 
 Metadata file: 
 Data Space Used: 2.356 GB
 Data Space Total: 49.92 GB
 Data Space Available: 47.57 GB
 Metadata Space Used: 946.2 kB
 Metadata Space Total: 1.615 GB
 Metadata Space Available: 1.614 GB
 Thin Pool Minimum Free Space: 3.494 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: true
 Deferred Deleted Device Count: 0
 Library Version: 1.02.107-RHEL7 (2016-06-09)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: host bridge null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 3.10.0-327.36.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 1.795 GiB
Name: ip-10-10-12-33
ID: FYYX:QRAH:7RLK:TDIV:3AKL:UHIQ:KPYF:SL5S:DWCU:XKPQ:O3XR:ZHPV
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
- AWS `t2.small` machine. EBS gp2 disks.
- Docker from `https://yum.dockerproject.org/repo/testing/centos/7/Packages/`
- One of the containers has `HEALTHCHECK`, the rest do not.
- `docker-compose.yml` includes `restart: always` for all containers

**Comment by thaJeztah** [comment]
ping @rhvgoyal @mlaventure PTAL

**Comment by rhvgoyal** [comment]
I suspect that this is device suspend/resume race which has been fixed by @shishir-a412ed

https://github.com/docker/docker/pull/23497

It would be good if we can backport it for 1.12.2

**Comment by thaJeztah** [comment]
I see there were concerns about including that change in a patch release (https://github.com/docker/docker/pull/23497#issuecomment-235364941)

@cpuguy83

**Comment by cpuguy83** [comment]
Yeah, it's a big patch and IIRC doesn't apply cleanly.

**Comment by cpuguy83** [comment]
@rhvgoyal Are you carrying this patch in RHEL?
Have you had many reports of this issue?

**Comment by rhvgoyal** [comment]
@cpuguy83, yes we have backported this fix in rhel docker 1.12

https://github.com/projectatomic/docker/pull/190

I think we have had 1-2 reports of this issue and that's why we debugged it and went on to fix it.

**Comment by rhatdan** [comment]
http://www.projectatomic.io/docs/docker_patches/

https://github.com/projectatomic/docker/tree/docker-1.12

**Comment by JamesJJ** [comment]
Issue exists in `docker-engine-1.12.3-0.0.20161025.001321.gitbad4d12.el7.centos.x86_64`

Please could you let us know if this is likely to be a target for 1.12.x or 1.13.x, or currently no/low priority?

**Comment by cpuguy83** [comment]
@JamesJJ 1.13

**Comment by Miyurz** [comment]
We have been facing this on our kubernetes cluster

 ```
Apr 29 20:56:43 ip-172-20-46-113 dockerd[982]: time="2017-04-29T20:56:43.998541501Z" level=debug msg="attach: stdout: begin"$
 Apr 29 20:56:43 ip-172-20-46-113 dockerd[982]: time="2017-04-29T20:56:43.998598366Z" level=debug msg="attach: stderr: begin"$ 
 Apr 29 20:56:44 ip-172-20-46-113 dockerd[982]: time="2017-04-29T20:56:44.188212698Z" level=warning msg="containerd: unable to save 41cd81912816f60de10d06683fbce615ca4b9f8438e7    db5aa9130b09d720124f:24548b991d6de47c20198e93363aff1983368aa36c220e1d3bd23896519b4b2c starttime: open /proc/22040/stat: no such file or directory"$
 Apr 29 20:56:44 ip-172-20-46-113 dockerd[982]: time="2017-04-29T20:56:44.190284772Z" level=info msg="containerd: 41cd81912816f60de10d06683fbce615ca4b9f8438e7db5aa9130b09d72012    4f:24548b991d6de47c20198e93363aff1983368aa36c220e1d3bd23896519b4b2c (pid 22040) has become an orphan, killing it"
```
Can someone throw some light on when this is going to be merged ? OR may be its already merged?

**Comment by roshangn88** [comment]
cc @Miyurz

**Comment by thaJeztah** [comment]
@Miyurz the linked PR (https://github.com/moby/moby/pull/23497) was merged for docker 1.13; consider upgrading to docker 17.03 or up. Looks like this issue was not closed after that PR was merged, so I'll close it now
